core/templates/pages/exploration-editor-page/services/user-email-preferences.service.ts

```diff
@@ -12,81 +12,94 @@
+    this.saveChangeToBackend({
+      message_type: this.MESSAGE_TYPE_SUGGESTION,
+      mute: mute
+    });
+  }
 
-angular.module('oppia').factory('UserEmailPreferencesService', [
-  '$http', 'AlertsService', 'ExplorationDataService',
-  'UrlInterpolationService',
-  function(
-      $http, AlertsService,
-      ExplorationDataService, UrlInterpolationService) {
-    var MESSAGE_TYPE_SUGGESTION = 'suggestion';
-    var MESSAGE_TYPE_FEEDBACK = 'feedback';
-    return {
-      init: function(
-          feedbackNotificationsMuted, suggestionNotificationsMuted) {
-        this.feedbackNotificationsMuted = feedbackNotificationsMuted;
-        this.suggestionNotificationsMuted = suggestionNotificationsMuted;
-      },
-      /**
-       * @return {boolean} Whether the feedback notification is muted.
-       */
-      areFeedbackNotificationsMuted: function() {
-        return this.feedbackNotificationsMuted;
-      },
-      /**
-       * @return {boolean} Whether the suggestion notification is muted.
-       */
-      areSuggestionNotificationsMuted: function() {
-        return this.suggestionNotificationsMuted;
-      },
-      /**
-       * Set the message type to feedback and mute to true or false.
-       * @param {boolean} mute - Whether the feedback notification is muted.
-       */
-      setFeedbackNotificationPreferences: function(mute) {
-        this.saveChangeToBackend({
-          message_type: MESSAGE_TYPE_FEEDBACK,
-          mute: mute
-        });
-      },
-      /**
-       * Set the message type to suggestion and mute to true or false.
-       * @param {boolean} mute - Whether the suggestion notification is muted.
-       */
-      setSuggestionNotificationPreferences: function(mute) {
-        this.saveChangeToBackend({
-          message_type: MESSAGE_TYPE_SUGGESTION,
-          mute: mute
-        });
-      },
-      /**
-       * Save the change of message_type and mute to backend.
-       * @param {object} requestParams - Info about message_type and mute.
-       */
-      saveChangeToBackend: function(requestParams) {
-        var that = this;
-        var emailPreferencesUrl = UrlInterpolationService.interpolateUrl(
-          '/createhandler/notificationpreferences/<exploration_id>', {
-            exploration_id: ExplorationDataService.explorationId
-          }
-        );
-        return $http.put(emailPreferencesUrl, requestParams).then(
-          function(response) {
-            var data = response.data;
-            AlertsService.clearWarnings();
-            that.init(
-              data.email_preferences.mute_feedback_notifications,
-              data.email_preferences.mute_suggestion_notifications);
-          }
-        );
-      }
-    };
+  /**
+   * Save the change of message_type and mute to backend.
+   * @param {RequestParam} requestParams - Info about message_type and mute.
+   */
+  saveChangeToBackend(requestParams: RequestParams) : Promise<void | object> {
```

Comment:
Hi @ashutoshc8101, I think what needs to be done is, you need to pass in a successCallback here and call that once `saveChangeToBackend` has resolved (i.e. at the end of all the statements inside the `then` block).

This successCallback will be a function that executes `$rootscope.$applyAsync()`. You will need to pass this function in all the places where `UserEmailPreferencesService.saveChangeToBackend` is called.

Reply:
Done.